.github/workflows/build.yaml

```diff
@@ -0,0 +1,43 @@
+name: Main build
+
+on:
+    workflow_dispatch:
+    pull_request:
+        branches: [main]
+
+jobs:
+    build-gradle:
+        name: build and Test Gradle 
+        runs-on: ubuntu-latest
+
+        steps:
+            - name: Checkout Code
+              uses: actions/checkout@v4
+            
+            - name: Setup JDK 
+              uses: actions/setup-java@v4
+              with:
+                java-version: '11'
+                distribution: 'temurin'
+            
+            - name: Setup Gradle
+              uses: gradle/actions/setup-gradle@v3
+              with:
+                gradle-version: 6.9.2
+  
+            - name: Test and Build Gradle
+              env:
+                GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+                ORG_GRADLE_PROJECT_signingKeyId:  ${{ secrets.GPG_KEYID }}
+                ORG_GRADLE_PROJECT_signingKey: ${{ secrets.GPG_SECRET }}

```

Comment:
Please change this to `GPG_KEY` as thats what our secret is named

Reply:
fixed